sky/provision/paperspace/instance.py

```diff
@@ -0,0 +1,297 @@
+        return common.ProvisionRecord(
+            provider_name='paperspace',
+            cluster_name=cluster_name_on_cloud,
+            region=region,
+            zone=None,
+            head_instance_id=head_instance_id,
+            resumed_instance_ids=list(newly_started_instances.keys()),
+            created_instance_ids=[],
+        )
+
+    created_instance_ids = []
+    for _ in range(to_start_count):
+        node_type = 'head' if head_instance_id is None else 'worker'
+        try:
+            instance_id = client.launch(
+                name=f'{cluster_name_on_cloud}-{node_type}',
+                instance_type=config.node_config['InstanceType'],
+                network_id=config.node_config['NetworkId'],
+                region=region,
+                disk_size=config.node_config['DiskSize'],
+            )['data']['id']
+        except Exception as e:  # pylint: disable=broad-except
+            logger.warning(f'run_instances error: {e}')
+            raise e
+        logger.info(f'Launched instance {instance_id}.')
+        created_instance_ids.append(instance_id)
+        if head_instance_id is None:
+            head_instance_id = instance_id
+
+    # Wait for instances to be ready.
+    while True:
+        instances = _filter_instances(cluster_name_on_cloud, ['ready'])
+        logger.info('Waiting for instances to be ready: '
+                    f'({len(instances)}/{config.count}).')
+        if len(instances) == config.count:
+            break
+
+        time.sleep(POLL_INTERVAL)
+    assert head_instance_id is not None, 'head_instance_id should not be None'
+    return common.ProvisionRecord(
+        provider_name='paperspace',
+        cluster_name=cluster_name_on_cloud,
+        region=region,
+        zone=None,
+        head_instance_id=head_instance_id,
+        resumed_instance_ids=list(stopped_instances.keys()),
+        created_instance_ids=created_instance_ids,
+    )
+
+
+def wait_instances(region: str, cluster_name_on_cloud: str,
+                   state: Optional[status_lib.ClusterStatus]) -> None:
+    del region, cluster_name_on_cloud, state  # unused
+    # We already wait on ready state in `run_instances` no need
+
+
+def stop_instances(
+    cluster_name_on_cloud: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+    worker_only: bool = False,
+) -> None:
+    del provider_config, worker_only  # unused
+    client = utils.PaperspaceCloudClient()
+    all_instances = _filter_instances(cluster_name_on_cloud, [
+        'ready', 'serviceready', 'upgrading', 'provisioning', 'starting',
+        'restarting'
+    ])
+    num_instances = len(all_instances)
+
+    # Request a stop on all instances
+    for instance_id in all_instances:
+        client.stop(instance_id=instance_id)
+
+    # Wait for instances to stop
+    for _ in range(MAX_POLLS_FOR_STOP):
+        all_instances = _filter_instances(cluster_name_on_cloud, ['off'])
+        if len(all_instances) < num_instances:
```

Comment:
```suggestion
        if len(all_instances) >= num_instances:
```

Should this wait all instances to be stopped?

Reply:
this is a silly bug 🙃 thanks for the catch